pkg/services/auth/auth.go

```diff
@@ -44,21 +44,38 @@ func NewAuthCLIHandler(name gitproviders.GitProviderName) (BlockingCLIAuthHandle
 // GetGitProvider returns a GitProvider containing either the token stored in the <git provider>_TOKEN env var
 // or a token retrieved via the CLI auth flow
 func GetGitProvider(ctx context.Context, url string) (gitproviders.GitProvider, error) {
-	osysClient := osys.New()
-
 	providerName, providerNameErr := gitproviders.DetectGitProviderFromUrl(url)
 	if providerNameErr != nil {
 		return nil, fmt.Errorf("error detecting git provider: %w", providerNameErr)
 	}
 
-	token, tokenErr := osysClient.GetGitProviderToken()
+	authHandler, authErr := NewAuthCLIHandler(providerName)
+	if authErr != nil {
+		return nil, fmt.Errorf("could not get auth handler for provider %s: %w", providerName, authErr)
+	}
+
+	osysClient := osys.New()
+	logger := logger.NewCLILogger(osysClient.Stdout())
+
+	return getGitProviderWithClients(ctx, providerName, osysClient, authHandler, logger)
```

Comment:
I only see the used once. Does it need to be a helper, or can we just do this work here in this function?

Reply:
I looked at it longer and realized this is a test-ability change.